keras_nlp/models/whisper/whisper_audio_feature_extractor.py

```diff
@@ -0,0 +1,222 @@
+# Copyright 2023 The KerasNLP Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import numpy as np
+import tensorflow as tf
+from tensorflow import keras
+
+NUM_MELS = 80
+
+
+class WhisperAudioFeatureExtractor(keras.layers.Layer):
+    """
+    Whisper audio feature extractor layer.
+
+    This layer takes in a batch of audio tensors, and computes the log-mel
+    spectrogram features for each audio tensor.
+
+    The input audio tensor can either be of shape `(length_of_audio,)` or
```

Comment:
For a follow up PR once we add presets, but if we can show a short enough example below of loading raw audio and bringing it to a format suitable for this layer that'd be very helpful to users.

Reply:
Possibly! One thing we should consider is if https://keras.io/api/data_loading/audio/ would work and could be extended to more file inputs.